### PR TITLE
Preview Generator Command: exclude & parent options

### LINKED ISSUE #8149: [Document] Preview Generator Command: exclude & parent options
It would be very helpful if you could specify a parent path for the generator (e.g. `/en`)  and additionally to exclude certain patterns (path regex).

### PR DESCRIPTION
**Pull request by mcop1**
Closes #8149

### CODE REVIEW

File: bundles/CoreBundle/Command/Document/GeneratePagePreviews.php [hunk 1]
```diff
@@ -43,9 +44,60 @@ protected function configure()
                 InputOption::VALUE_OPTIONAL,
                 'Prefix for the document path, eg. https://example.com, if not specified, Pimcore will try use the main domain from system settings.',
                 null
+            )
+            ->addOption(
+                'parent',
+                'p',
+                InputOption::VALUE_OPTIONAL,
+                'Only create preview of objects in this folder (ID|path).'
```
Comment (dvesh3) [inline bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]:
```suggestion
                'Only create preview of documents in parent hierarchy (ID|path).'
```

File: bundles/CoreBundle/Command/Document/GeneratePagePreviews.php [hunk 2]
```diff
@@ -43,9 +44,60 @@ protected function configure()
...
+            )
+            ->addOption(
+                'regex',
+                'r',
```
Comment (dvesh3) [inline bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]: I think it is more clear
```suggestion
                'exclude-patterns',
                'e',
```

File: bundles/CoreBundle/Command/Document/GeneratePagePreviews.php [hunk 3]
```diff
@@ -43,9 +44,60 @@ protected function configure()
...
+                InputOption::VALUE_OPTIONAL,
+                'All child objects matching the regex pattern will be excluded.'
```
Comment (dvesh3) [inline bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]:
```suggestion
                'All documents matching the regex pattern will be excluded.'
```

File: bundles/CoreBundle/Command/Document/GeneratePagePreviews.php [hunk 4]
```diff
@@ -43,9 +44,60 @@ protected function configure()
...
+            }
+            if ($parent instanceof Document) {
+                $conditions[] = "path LIKE " . $db->quote($parent->getRealFullPath() . "%");
+            } else {
+                $this->writeError($parentIdOrPath . ' is not a valid id or path!');
+                exit(1);
+            }
+        }
+
+        $regex = $input->getOption('regex');
```
Comment (dvesh3) [inline bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]:
```suggestion
        $regex = $input->getOption('exclude-patterns');
```

File: bundles/CoreBundle/Command/Document/GeneratePagePreviews.php (hunk 3 above)
Comment (mcop1) [reply bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]: The regex tests the path property, so maybe we should even clarify it further? Something like "Excludes all documents whose path property matches the regex pattern"?
What do you think?

Comment (dvesh3) [reply bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]: yes, even better 👍

File: bundles/CoreBundle/Command/Document/GeneratePagePreviews.php [hunk 5]
```diff
@@ -43,9 +44,60 @@ protected function configure()
...
+            )
+            ->addOption(
+                'parent',
+                'p',
+                InputOption::VALUE_OPTIONAL,
+                'Only create preview of documents in parent hierarchy (ID|path).'
+            )
+            ->addOption(
+                'exclude-patterns',
+                'e',
```
Comment (dvesh3) [inline bundles/CoreBundle/Command/Document/GeneratePagePreviews.php]: we cannot use option "e" as it is already being used for environment variable 
image
```suggestion
                'x',
```

**Pull request merged**